src/main/resources/db/migration/V0.1.1__usage_interval_entities.sql

```diff
@@ -0,0 +1,50 @@
+DO $$
+BEGIN
+  IF NOT EXISTS (SELECT FROM pg_type t INNER JOIN pg_namespace ns ON t.typnamespace = ns.oid WHERE t.typname = 'quality_of_reading' AND ns.nspname = 'usage') THEN
+    CREATE TYPE usage.quality_of_reading AS ENUM (
+      'VALID',
+      'MANUALLY_EDITED',
+      'ESTIMATED_USING_REFERENCE_DAY',
+      'ESTIMATED_USING_LINEAR_INTERPOLATION',
+      'QUESTIONABLE',
+      'DERIVED',
+      'PROJECTED_FORECAST',
+      'MIXED',
+      'RAW',
+      'NORMALIZED_FOR_WEATHER',
+      'OTHER',
+      'VALIDATED',
+      'VERIFIED',
+      'REVENUE_QUALITY'
+    );
+  END IF;
+END $$;
+
+CREATE TABLE IF NOT EXISTS usage.interval_block (
+  uuid UUID PRIMARY KEY,
+  description TEXT,
+  published TIMESTAMP,
+  self_link_href TEXT,
+  self_link_rel TEXT,
+  up_link_href TEXT,
+  up_link_rel TEXT,
+  updated TIMESTAMP
+);
+
+CREATE TABLE IF NOT EXISTS usage.interval_reading (
+  id BIGSERIAL PRIMARY KEY,
+  cost BIGINT,
+  start BIGINT,
+  duration INTEGER,
+  value BIGINT,
+  consumption_tier SMALLINT,
+  tou SMALLINT,
+  cpp SMALLINT,
+  block_uuid UUID NOT NULL REFERENCES usage.interval_block ON DELETE CASCADE
```

Comment:
I like to add `ON DELETE CASCADE` on the many side for isolated one-to-many relationships (e.g., if an `IntervalBlock` is deleted, there's no reason to keep around the related `IntervalReading`s), but I can remove those clauses if you prefer.

Reply:
You are correct.  The IntervalReading should have a foreign key to the related IntervalBlock (this is missing in the .mwb files) so cascading the deletion is the correct thing to do.